main/modes/games/cGrove/Sparring/cg_Match.c

```diff
@@ -0,0 +1,1025 @@
+{
+    // Initialize
+    // Load tournament/match data
+    strcpy(cg->spar.match.matchName, matchName);
+    cg->spar.match.round    = round;
+    cg->spar.match.maxTime  = maxTime;
+    cg->spar.match.animDone = false;
+    cg->spar.match.done     = false;
+
+    // Timer is set to 0
+    cg->spar.match.timer = 0;
+
+    // AI
+    cg->spar.match.ai.aiDifficulty = ai;
+
+    // Chowa
+    cg->spar.match.chowaData[CG_P1].chowa = player1Chowa;
+    cg->spar.match.chowaData[CG_P2].chowa = player2Chowa;
+    for (int32_t i = 0; i < 2; i++)
+    {
+        // FIXME: Setting stats for test purposes
+        cg->spar.match.chowaData[i].chowa->stats[CG_SPEED]    = 128;
+        cg->spar.match.chowaData[i].chowa->stats[CG_STAMINA]  = 128;
+        cg->spar.match.chowaData[i].chowa->stats[CG_STRENGTH] = 0;
+        cg->spar.match.chowaData[i].chowa->stats[CG_AGILITY]  = 128;
+        cg->spar.match.chowaData[i].chowa->stats[CG_HEALTH]   = 128;
+        cg->spar.match.chowaData[i].chowa->playerAffinity     = 255;
+        // Charisma not needed for spar
+
+        cg->spar.match.chowaData[i].currState = CG_SPAR_UNREADY;
+        cg->spar.match.chowaData[i].currMove  = CG_SPAR_UNSET;
+        cg->spar.match.chowaData[i].maxStamina
+            = MIN_STAMINA + (cg->spar.match.chowaData[i].chowa->stats[CG_STAMINA] >> 1);
+        cg->spar.match.chowaData[i].stamina   = cg->spar.match.chowaData[i].maxStamina;
+        cg->spar.match.chowaData[i].readiness = 0;
+        cg->spar.match.chowaData[i].readiness = 0;
+        cg->spar.match.chowaData[i].maxHP
+            = MIN_HP + cg->spar.match.chowaData[i].chowa->stats[CG_HEALTH] * 3; // 250 - 1000
+        cg->spar.match.chowaData[i].HP          = cg->spar.match.chowaData[i].maxHP;
+        cg->spar.match.chowaData[i].updateTimer = 0;
+    }
+}
+
+/**
+ * @brief Executes the logic for the spar
+ *
+ * @param cg Game data
+ * @param elapsedUs time since last frame
+ */
+void cg_runSparMatch(cGrove_t* cg, int64_t elapsedUs)
+{
+    // Check if match has ended
+    if (cg->spar.match.done == true)
+    {
+        // TODO: wait until animations finish, then save data and load next match or show a summary
+        return;
+    }
+
+    // Round timer
+    // Timer doesn't accumulate if paused and single player
+    if (!cg->spar.match.paused && !cg->spar.match.online)
+    {
+        cg->spar.match.usTimer += elapsedUs;
+    }
+    // If enough us have built up, increments seconds timer
+    if (cg->spar.match.usTimer >= SECOND)
+    {
+        cg->spar.match.usTimer -= SECOND;
+        cg->spar.match.timer += 1;
+    }
+    // If enough seconds have passed, end match
+    if (cg->spar.match.timer >= cg->spar.match.maxTime)
+    {
+        cg->spar.match.done        = true;
+        cg->spar.match.finalResult = CG_DRAW;
+        return;
+    }
+
+    // Loop over both Chowa
+    cg_sparMatchPlayerInput(cg);
+    if (cg->spar.match.online)
+    {
+        // TODO: Get input from other swadge
```

Comment:
Didn't realize this was P2P

Reply:
I mean, I'd like that if possible. I'm steadily running out of time though.

Reply:
Yeah, P2P almost always ends up being more trouble than you realize. I'm ok leaving this on the cutting room floor